### DISCUSSION TITLE
Calling dict() on some ak.Array object

### DISCUSSION THREAD

**Discussion by HenryDayHall** [scikit-hep/awkward#798] [Q&A]
So sometimes calling `dict(ak_array)` gives an error.

```python

In [10]: import awkward as ak
    ...: print(ak.__version__)
    ...: dog = ak.from_iter([[1., 2.], [5.]])
    ...: pets = ak.zip({"dog": dog[np.newaxis]}, depth_limit=1)
1.1.2

In [11]: dict(pets)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-11-f7ecb9a598b4> in <module>
----> 1 dict(pets)

~/Programs/anaconda3/envs/tree/lib/python3.7/site-packages/awkward/highlevel.py in __getitem__(self, where)
   1765             2
   1766         """
-> 1767         return ak._util.wrap(self._layout[where], self._behavior)
   1768
   1769     def __setitem__(self, where, what):

ValueError: scalar Record can only be sliced by field name (string); try "0"

(https://github.com/scikit-hep/awkward-1.0/blob/1.1.2/src/libawkward/array/Record.cpp#L198)

```
It is more than possible I have done something odd with my packages, so let me know if you don't reproduce it. Also apologies if it's already been reported.

Side note for future readers; if you got here with the same goal I had then you wanted the list of column names, it's `ak.fields(pets)`.

**Comment by jpivarski** [comment]
I don't know what you want `dict(ak_array)` to do. Technically, what happens is that the `dict` constructor iterates over what it's given and tries to cast each item as a 2-tuple to use as key-value pairs. Some Awkward Arrays will have that structure and some won't.

Are you wanting to turn the Awkward Array into Python lists and dicts? That's [ak.to_list](https://awkward-array.readthedocs.io/en/latest/_auto/ak.to_list.html) or `ak_array.tolist()` as a method (following NumPy's [naming convention](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html)).

```python
>>> pets
<Array [{dog: [[1, 2], [5]]}] type='1 * {"dog": var * var * float64}'>
>>> pets.tolist()
[{'dog': [[1.0, 2.0], [5.0]]}]
```

**Reply by HenryDayHall** [reply]
Oh that's cool, yes, that is what I was hopping to get. Thanks.
